amethyst_ui/src/prefab.rs

```diff
@@ -282,3 +293,242 @@ where
         self.image.trigger_sub_loading(progress, &mut system_data.1)
     }
 }
+
+/// Loadable ui components
+///
+/// ### Type parameters:
+///
+/// - `I`: `Format` used for loading `Texture`s
+/// - `F`: `Format` used for loading `FontAsset`
+#[derive(Serialize, Deserialize)]
+pub enum UiWidget<I, F>
+where
+    I: Format<Texture, Options = TextureMetadata>,
+    F: Format<FontAsset, Options = ()>,
+{
+    /// Container component, have a transform, an optional background image, and children
+    Container(
+        UiTransformBuilder,
+        Option<UiImageBuilder<I>>,
+        Vec<UiWidget<I, F>>,
+    ),
+    /// Image component, have a transform and an image
+    Image(UiTransformBuilder, UiImageBuilder<I>),
+    /// Text component, have a transform and text
+    Text(UiTransformBuilder, UiTextBuilder<F>),
+    /// Button component, have a transform, an image and text
+    Button(UiTransformBuilder, UiImageBuilder<I>, UiTextBuilder<F>),
+}
+
+type UiPrefabData<I, F> = (
+    Option<UiTransformBuilder>,
+    Option<UiImageBuilder<I>>,
+    Option<UiTextBuilder<F>>,
+);
+
+/// Ui prefab
+///
+/// ### Type parameters:
+///
+/// - `I`: `Format` used for loading `Texture`s
+/// - `F`: `Format` used for loading `FontAsset`
+pub type UiPrefab<I, F> = Prefab<UiPrefabData<I, F>>;
+
+/// Ui format.
+///
+/// Load `UiPrefab` from `ron` file.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UiFormat;
+
+impl<I, F> SimpleFormat<UiPrefab<I, F>> for UiFormat
+where
+    I: Format<Texture, Options = TextureMetadata> + Sync + for<'a> Deserialize<'a>,
+    F: Format<FontAsset, Options = ()> + Sync + for<'a> Deserialize<'a>,
+{
+    const NAME: &'static str = "Ui";
```

Comment:
Correct me if I'm wrong, but I think we used a path for this so far.

Reply:
No, the others have single word names.

Reply:
`const NAME: &'static str = "OGG";` as an example